pkg/secrethub/internals/http/client.go

```diff
@@ -245,11 +245,40 @@ func (c *Client) DeleteRepo(namespace, repoName string) error {
 }
 
 // AuditRepo gets the audit events for a given repo.
-func (c *Client) AuditRepo(namespace, repoName string, subjectTypes api.AuditSubjectTypeList) ([]*api.Audit, error) {
-	out := []*api.Audit{}
-	rawURL := fmt.Sprintf(pathRepoEvents+"?subject_types=%s", c.base, namespace, repoName, subjectTypes.Join(","))
-	err := c.get(rawURL, true, &out)
-	return out, errio.Error(err)
+func (c *Client) AuditRepo(namespace, repoName string, subjectTypes api.AuditSubjectTypeList) (*Paginator, error) {
```

Comment:
Now that we're changing this signature, let's also change the way we pass filter options such as the subject types, so that we don't change the signature again when adding more filter options.

Reply:
Let's discuss first whether we actually want to break the signature. Alternatively we could e.g. add a new function and deprecate the old one.

Reply:
Good question :+1:
I think this also depends on how we add pagination on the API. For audit events we have done so on the existing route and by doing so changed the behavior of the `ListEvents` function. It now limits it's results to the first 50 events.
Because of this change in behavior, I feel more for dropping the function altogether.

For other routes, we can deprecate first before removing the list functions. We can already add the functions that return iterators before adding pagination on those routes (we just iterate over one page that contains all entries). We can then remove the function before adding pagination:
add iterator funcs => deprecate list funcs => :hourglass_flowing_sand: => remove list funcs => add pagination on routes

Therefore, I propose to change the function name of the audit iterators regardless of whether we want the list functions to keep the same signature. So that we can follow the same naming pattern for the other routes.

How about `Iterator` for a function name?
e.g. `client.Users().Members().Iterator()` vs `client.Users().Members().List()`
`client.Users().Repo().EventIterator()` vs `client.Users().Repo().ListEvents()`

I like that it signals that nothing happens yet. `List` is active, it does something (retrieves the events from the API), `Iterator` just returns something with which you can do something (iterating).